atmcfg/atmcfg_test.go

```diff
@@ -140,6 +160,46 @@ func TestAddUser(t *testing.T) {
 	}
 }
 
+func TestAddIndexConfig(t *testing.T) {
+	config := automationConfigWithIndexConfig()
+	index := &opsmngr.IndexConfigs{
+		DBName:         "test1",
+		CollectionName: "test2",
+		RSName:         "test2",
+		Key: [][]string{
+			{
+				"test", "test",
+			},
+		},
+		Options:   nil,
+		Collation: nil,
+	}
+
+	var err error
+
+	t.Run("AutomationConfig not initialized", func(t *testing.T) {
+		err = AddIndexConfig(nil, index)
+		if err == nil {
+			t.Error("AddIndexConfig did not throw an exception")
+		}
+	})
+
+	t.Run("add an index", func(t *testing.T) {
+		err = AddIndexConfig(config, index)
+		if err != nil || len(config.IndexConfigs) != 2 {
+			t.Error("indexConfig has not been added to the AutomationConfig")
+		}
+	})
+
+	t.Run("trying to add the same index in the AutomationConfig", func(t *testing.T) {
+		err = AddIndexConfig(config, index)
```

Comment:
this test depends on the previous test running, if we need to refactor the tests or change the order this test may start to fail, please define the `config` inside the scope of the scenario and not as a shared dependency (because of the previous point, multiple test modifying the same input will make your tests brittle later on) this also mean that for this particular test you may need to run add multiple times